test/algorithms/relational_operations/relate/test_relate.hpp

```diff
@@ -67,6 +70,44 @@ std::string matrix_format(std::string const& matrix1, std::string const& matrix2
          + ( !matrix2.empty() ? " || " : "" ) + matrix2;
 }
 
+template <typename M>
+char get_ii(M const& m)
```

Comment:
Indeed. Just checked test_buffer and there it is never done...
In general I think a free function in any hpp should be inlined, so maybe I will change buffer too.
But as long as the hpp is used in one cpp at at time it should be no problem.